modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java

```diff
@@ -1672,4 +1675,46 @@ private static String getAssociatedDocFileName(String filepath) {
     //            e.printStackTrace();
     //        }
     //    }
+
+    public String setPermission(String userName, String processName, String processVersion) throws ProcessCenterException {
+
+        String status="Failed to set permission";
+
+        try {
+
+            RegistryService registryService = ProcessCenterServerHolder.getInstance().getRegistryService();
+
+            if (registryService != null) {
+                UserRegistry userRegistry = registryService.getGovernanceSystemRegistry();
+                UserRealm userRealm = userRegistry.getUserRealm();
+                String[] roles=userRealm.getUserStoreManager().getRoleListOfUser(userName);
+
+                String path = "/_system/governance/processes/" + processName + "/" +
+                        processVersion;
+
+                for(String role:roles){
+
+                    if(role.equalsIgnoreCase("Internal/everyone")||role.equalsIgnoreCase("Internal/store")||role.equalsIgnoreCase("Internal/publisher")){
+                        continue;
+                    }
+                    else {
+                        //add read permission
+                        AddRolePermissionUtil.addRolePermission(userRegistry, path, role, ProcessCenterConstants.READ, ProcessCenterConstants.ALLOW);
+                        //add write permission
+                        AddRolePermissionUtil.addRolePermission(userRegistry, path, role, ProcessCenterConstants.WRITE, ProcessCenterConstants.ALLOW);
+                        //add authorize permission
+                        AddRolePermissionUtil.addRolePermission(userRegistry, path, role, ProcessCenterConstants.AUTHORIZE, ProcessCenterConstants.ALLOW);
+                    }
+                }
+                status="Permission set successfully";
+            }
+        }catch (Exception e) {
+            String errMsg = "Failed to update Permission" ;
+            log.error(errMsg, e);
```

Comment:
No need to do both logging and throwing the exception. This will cause duplicate error logs.